clarity/src/vm/clarity_wasm.rs

```diff
@@ -1566,9 +1554,9 @@ fn pass_argument_to_wasm(
             buffer.push(Val::I32(written));
             Ok((buffer, offset + written, in_mem_offset + in_mem_written))
         }
-        Value::Principal(_p) => todo!("Value type not yet implemented: {:?}", value),
-        Value::CallableContract(_c) => todo!("Value type not yet implemented: {:?}", value),
-        Value::Tuple(_t) => todo!("Value type not yet implemented: {:?}", value),
+        Value::Principal(_p) => Err(Error::Wasm(WasmError::UnhandledType)),
+        Value::CallableContract(_c) => Err(Error::Wasm(WasmError::UnhandledType)),
+        Value::Tuple(_t) => Err(Error::Wasm(WasmError::UnhandledType)),
```

Comment:
Same for these, but don't worry about opening an issue for these, @Acaccia is already working on them.

Reply:
Ok. Reverted.